Source/JavaScriptCore/dfg/DFGSpeculativeJIT64.cpp

```diff
@@ -3537,6 +3537,8 @@ void SpeculativeJIT::compile(Node* node)
             bool canIgnoreNegativeZero = bytecodeCanIgnoreNegativeZero(node->arithNodeFlags());
             branchConvertDoubleToInt52(valueFPR, resultGPR, failureCases, scratch1GPR, scratch2FPR, canIgnoreNegativeZero);
 
+            if (canIgnoreNegativeZero)
+                node->child1().setUseKind(DoubleRepRealUse);
```

Comment:
We should not change the useKind edge in code generator. This is only allowed in a certain places including Constant Folding / Strength Reduction etc.

Reply:
Can we add an assertion for that?

Reply:
Personally I don't like that since it makes all code much more complicated.